search/search_repository.go

```diff
@@ -225,16 +226,24 @@ func parseSearchString(rawSearchString string) searchKeyword {
 		// TODO: need to find out the way to use ID fields.
 		if strings.HasPrefix(part, "id:") {
 			res.id = append(res.id, strings.TrimPrefix(part, "id:")+":*A")
+		} else if strings.HasPrefix(part, "type:") {
+			typeName := strings.TrimPrefix(part, "type:")
+			if len(typeName) == 0 {
+				return res, models.NewBadParameterError("Type name must not be empty", part)
+			}
+			res.types = append(res.types, typeName)
 		} else if govalidator.IsURL(part) {
-			part := trimProtocolFromURLString(part)
+			part := strings.ToLower(part)
```

Comment:
I've moved the "toLowercase" from the mainline code to the places where I think it is not wrong (minimal change). Generally, I don't think it makes sense to lowercase search strings: that is the job of the to_tsvector and to_tsquery functions (stemming to lexemes). As for matching URL's, that could be done by writing regexes that are case-insensitive. However, I think this is outside of the scope of this PR.